### PR TITLE
refactor(insights): use query based insights in `<InsightDetails />`

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

We'd like to move from `filters` to `query`.

## Changes

This uses a frontend-side conversion function to use `query`-nodes in the `<InsightDetails />` component and related sub-components.

## How did you test this code?

Used a test dashboard

### CODE REVIEW

File: frontend/src/lib/components/Cards/InsightCard/TopHeading.tsx [hunk 1]
```diff
@@ -23,7 +24,7 @@ export function TopHeading({ insight }: { insight: QueryBasedInsightModel }): JS
         }
     } else {
         // maintain the existing default
-        insightType = INSIGHT_TYPES_METADATA[InsightType.TRENDS]
+        insightType = QUERY_TYPES_METADATA[NodeKind.TrendsQuery]
```
Comment (thmsobrmlr) [inline frontend/src/lib/components/Cards/InsightCard/TopHeading.tsx:27]: Unrelated change that I'm smuggling in with this PR

**Review by webjunkie — APPROVED** [review]
